src/modules/admin/components/services/ServiceTypeDialog.tsx

```diff
@@ -0,0 +1,197 @@
+    }
+  }, [pickList, serviceType, dialogOpen]);
+
+  const onClose = useCallback(() => {
+    closeDialog();
+    // Don't reset to defaults here - that would cause the update dialog to show null values on reopen
+    setErrors(emptyErrorState);
+  }, [setErrors, closeDialog]);
+
+  const onMutationCompleted = useCallback(
+    (errors: ValidationError[] = []) => {
+      if (errors.length > 0) {
+        setErrors(partitionValidations(errors));
+      } else {
+        evictQuery('serviceTypes');
+        evictQuery('pickList', {
+          pickListType: PickListType.CustomServiceCategories,
+        });
+        onClose();
+      }
+    },
+    [setErrors, onClose]
+  );
+
+  const [updateServiceType, { error: updateError, loading: updateLoading }] =
+    useUpdateServiceTypeMutation({
+      variables: {
+        id: serviceType?.id || '', // will always be present if update mutation is called
+        input: mutationInput,
+      },
+      onCompleted: (data: UpdateServiceTypeMutation) => {
+        onMutationCompleted(data.updateServiceType?.errors);
+      },
+    });
+
+  const [createServiceType, { error: createError, loading: createLoading }] =
+    useCreateServiceTypeMutation({
+      variables: {
+        input: mutationInput,
+      },
+      onCompleted: (data: CreateServiceTypeMutation) => {
+        onMutationCompleted(data.createServiceType?.errors);
+        if (data.createServiceType?.serviceType?.id) {
+          navigate(
+            generateSafePath(AdminDashboardRoutes.CONFIGURE_SERVICE_TYPE, {
+              serviceTypeId: data.createServiceType?.serviceType?.id,
+            })
+          );
+        }
+      },
+    });
+
+  if (createError) throw createError;
+  if (updateError) throw updateError;
+  if (pickListError) throw pickListError;
+
+  return (
+    <CommonDialog open={dialogOpen}>
+      <DialogTitle>{serviceType ? 'Edit' : 'Create'} Service Type</DialogTitle>
+      <DialogContent sx={{ mt: 2 }}>
+        <Stack gap={2}>
+          {errors.errors.length > 0 && (
+            <Alert severity='error'>
+              <AlertTitle>Failed to submit</AlertTitle>
+              <ValidationErrorList errors={errors.errors} />
+            </Alert>
+          )}
+          <TextInput
+            label='Name'
+            value={name}
+            sx={{ width: 500, maxWidth: '100%' }}
+            onChange={(e) => setName(e.target.value)}
+          />
+          <LabeledCheckbox
+            label='Supports bulk assignment'
+            value={supportsBulkAssignment}
+            onChange={(_e, checked) => setSupportsBulkAssignment(checked)}
+          />
+          <CreatableFormSelect
+            value={serviceCategory}
+            options={pickList || []}
+            onChange={(_e, value) => {
+              setServiceCategory(isPickListOption(value) ? value : null);
+            }}
+            loading={pickListLoading}
+            placeholder='Service Category'
```

Comment:
```suggestion
            placeholder='Select Service Category'
```

following typical form input pattern in this app (`placeholderText` helper fn used in dynamic forms)